hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/format/mor/MergeOnReadInputFormat.java

```diff
@@ -135,6 +139,12 @@
    */
   private boolean closed = true;
 
+  private final Option<SchemaEvolutionContext> schemaEvolutionContext;
+  private List<String> actualFieldNames;
+  private List<DataType> actualFieldTypes;
+  private InternalSchema actualSchema;
+  private InternalSchema querySchema;
```

Comment:
Same rule, just make the schema correct before constructing the `MergeOnReadInputFormat`, and let's not impose the components of schema evolution in any format/reader.

Reply:
There is no correct schema at moment of contraction `MergeOnReadInputFormat`:
baseFile1 with schema1 {id: int, value: int}
baseFile2 with schema2 {id: int, value: long} -- read schema
both files will be passed to `MergeOnReadInputFormat#open` as `MergeOnReadInputSplit`
we need to read baseFile1 with schema1 then cast `value: int` to `value: long` using `CastMap`

Reply:
Sorry I don't get what do you mean by schema metadata that tells us the schema
My point is that to read baseFile1 we need to use `DataType[] fullFieldTypes` in `ParquetSplitReaderUtil.genPartColumnarRowReader` according to schema1

Reply:
Fields `actualFieldNames`, `actualFieldTypes`, `actualSchema`, `querySchema` are removed